openfl/utilities/optimizers/torch/fedprox.py

```diff
@@ -115,34 +172,53 @@ def step(self, closure=None):
             nesterov = group["nesterov"]
             mu = group["mu"]
             w_old = group["w_old"]
-            for p, w_old_p in zip(group["params"], w_old):
+
+            # Validate old weights for FedProx
+            self._validate_old_weights(mu, w_old)
+
+            # Apply proximal term when mu != 0
+            apply_proximal = w_old is not None
```

Comment:
Overall, consider a simpler approach: 

`apply_proximal` should always be true. We guarantee that `mu` will be `>=0.0` during initialization. Which means negative values never appear. As for `0.0` case, it means "no contribution" of the regularizer term. It becomes implied that it makes no effect on the weights.